src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftToken.java

```diff
@@ -0,0 +1,37 @@
+package io.fabric8.jenkins.openshiftsync;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import hudson.Extension;
+import hudson.util.Secret;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+// TODO: merge with https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/OpenShiftTokenCredentialImpl.java  ?
+public class OpenShiftToken extends BaseStandardCredentials {
+
+    private final Secret secret;
+
+    @DataBoundConstructor
+    public OpenShiftToken(CredentialsScope scope, String id, String description, Secret secret) {
+        super(scope, id, description);
+        this.secret = secret;
+    }
+
+    public String getToken() {
+        return secret.getPlainText();
```

Comment:
We would just need to add the class of the Username/Password credential object to the list population process [here](https://github.com/oatmealraisin/jenkins-sync-plugin/blob/860ee10041965b93690c9605aafaa7c4903ab0ae/src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java#L138), and then accommodate for it when we look for it in [getCurrentToken](https://github.com/oatmealraisin/jenkins-sync-plugin/blob/860ee10041965b93690c9605aafaa7c4903ab0ae/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java#L68)

I'm assuming the secret is the plaintext that the user provides, since the kubernetes plugin knows what to do with it. Confirming that atm..